src/auth/store/reducers/index.ts

```diff
@@ -1 +1,66 @@
-export * from './auth.reducer';
+import {UserInterface, UserModel} from '../../models/user.model';
+import { AuthActionTypes, AuthActions } from '../actions/auth.actions';
+import {createFeatureSelector} from '@ngrx/store';
+
+export interface AuthState {
+  // is a user authenticated?
+  isAuthenticated: boolean;
+  // if authenticated, there should be a user object
+  user: UserModel | null;
+  userDetailsLoaded: boolean;
+  userDetailsLoading: boolean;
+  permissions: string;
+  // error messages
+  errors: { [id: string]: string };
+}
+
+export const initialState: AuthState = {
+  isAuthenticated: false,
+  user: null,
+  userDetailsLoaded: false,
+  userDetailsLoading: false,
+  permissions: '',
+  errors: {
+    forgotPass: '',
+    login: ''
+  },
+};
+
+export function reducer (
+  state = initialState,
+  action: AuthActions
+): AuthState {
+  switch (action.type) {
+    case AuthActionTypes.LOGIN_SUCCESS: {
+      return {
+        ...state,
+        isAuthenticated: true,
+        permissions: action.payload.data,
+        errors: null
+      };
+    }
+
+    case AuthActionTypes.USER_SUCCESS: {
+      const user = new UserModel(action.payload);
+      return {
+        ...state,
+        user,
+        userDetailsLoaded: true,
+        userDetailsLoading: false,
```

Comment:
think we could keep this as loaded loading just for a sake of consistency... don't think we need to be explicit

Reply:
yeah good point